packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp

```diff
@@ -147,22 +147,20 @@ namespace Ifpack2 {
     template<typename T, int N>
     static 
     KOKKOS_INLINE_FUNCTION 
-    volatile ArrayValueType<T,N>& 
+    void
     operator+=(volatile ArrayValueType<T,N> &a, 
                volatile const ArrayValueType<T,N> &b) {
       for (int i=0;i<N;++i) 
         a.v[i] += b.v[i];
-      return a;
     }
     template<typename T, int N>
     static 
     KOKKOS_INLINE_FUNCTION
-    ArrayValueType<T,N>& 
+    void
```

Comment:
You don't need to change this one.  (I suppose it's harmless, unless you want to be able to chain operators.)

Reply:
Ummm..... I just want to keep the code at the same manner. You are right that the operator would not be used for chaining but the warning should be removed in building the code anyway.

Reply:
I'm not too worried about this -- if people ever need to chain `ArrayValueType` `operator+=`, we could always add the return type back in.

Reply:
This is not a Scalar type that we intend to expose for general use, right?

Reply:
No, it is not. I just want to do multiple reductions in a single parallel for using a fixed array type to reduce the number of parallel for launching and multiple memory access. If Kokkos supports this kind of types and its reduction operator, it would be nice though.